src/ucp/rndv/rndv_get.c

```diff
@@ -128,6 +179,37 @@ ucp_proto_rndv_get_zcopy_query(const ucp_proto_query_params_t *params,
     ucp_proto_rndv_bulk_query(params, attr);
 }
 
+static UCS_F_ALWAYS_INLINE size_t ucp_proto_rndv_get_zcopy_max_payload(
+        ucp_request_t *req, const ucp_proto_rndv_bulk_priv_t *rpriv,
+        const ucp_proto_multi_lane_priv_t *lpriv, ucp_lane_index_t *lane_shift)
+{
+    size_t total_offset = ucp_proto_rndv_request_total_offset(req);
+    size_t iov_index, iov_length;
+    size_t max_payload;
+
+    max_payload = ucp_proto_rndv_bulk_max_payload_align(req, rpriv, lpriv,
+                                                        lane_shift);
+    if (req->send.rndv.rma_count == 0) {
+        return max_payload;
+    }
+    iov_index = req->send.rndv.rma_index;
```

Comment:
What test and with what parameters should I run to reach this point? I tried `test_ucp_am_nbx_rndv.dts` but it always chooses other protos.

Reply:
If use `UCX_PROTO_ENABLE=y UCX_RNDV_SCHEME=get_zcopy UCX_RNDV_THRESH=0`, it will choose `rndv_get` for iov definitely. And if the message length is zero, it will fail when register memory handler and cause other unexpected behaviors.

Reply:
If use `UCX_PROTO_ENABLE=y UCX_RNDV_SCHEME=get_zcopy UCX_RNDV_THRESH=0`, it will choose `rndv_get` for iov definitely.

Reply:
Is it possible to create very simple and straightforward test-case expressed just in several lines that covers this new flow?

E.g. add test case to `test_ucp_am_nbx_rndv` with implied `UCX_PROTO_ENABLE=y UCX_RNDV_SCHEME=get_zcopy UCX_RNDV_THRESH=0` environment.

The diff in this patch is already around 500 lines so I would like to include testing in that patch only if it is very straightforward and simple change.

Reply:
done